crates/ruff_python_ast/src/str.rs

```diff
@@ -67,15 +67,55 @@ pub fn is_triple_quote(content: &str) -> bool {
     TRIPLE_QUOTE_STR_PREFIXES.contains(&content) || TRIPLE_QUOTE_BYTE_PREFIXES.contains(&content)
 }
 
+/// Return `true` if the string expression is an implicit concatenation.
+pub fn is_implicit_concatenation(content: &str) -> bool {
+    let leading_quote_str = leading_quote(content).unwrap();
+    let trailing_quote_str = trailing_quote(content).unwrap();
```

Comment:
Can we return `false` if the string does not have any quotes (avoids the potential panic)
```suggestion
    let Some(leading_quote_str) = leading_quote(content) else { return false; }
    let Some(trailing_quote_str) = trailing_quote(content) else { return false; }
```